### DISCUSSION TITLE
References view - is it working?  Or, how to use

### DISCUSSION THREAD

**Discussion by Gregory1701** [codefori/vscode-ibmi#1373] [IBM i] [answered]
When in a simple program, I right click on the variable thisPgm and select Find all references.  This works:
![image](https://github.com/halcyon-tech/vscode-ibmi/assets/132368707/47c22899-84d3-48bf-8d24-5f816bbb4bd7)

However, then I click on the IBM i extension and open a more complicated program and put my cursor on the variable idx and try to show all references again.  It says no results.
![image](https://github.com/halcyon-tech/vscode-ibmi/assets/132368707/02ebf1ab-3ffe-4908-bb86-8e9f458b4e58)
I guess it's trying to be helpful in suggesting that I try running a previous search again.  But why would I want to try running a search on the thisPgm variable?  I'm looking for all references to the idx variable.  Although, from the above screen shot, you can see it's used a lot.  This variable, idx, is also a local stand alone variable defined within this same procedure.  I'm trying to understand if this is a bug or I'm potentially doing something wrong.

**Comment by worksofliam** [comment]
@Gregory1701 please also share the code so we can copy and paste into our own instances. Screenshots don't help us too much for debugging!

**Reply by Gregory1701** [reply, marked as answer]
I found the difference.  It works if the program is **FREE.  No references found if the RPG is not **FREE.

       Ctl-Opt Main(Proc_Name);
       Ctl-Opt Debug Option(*SrcStmt:*NoDebugIO);
       Ctl-Opt ActGrp(*Caller);
       Ctl-opt ExtBinInt(*Yes);

       Dcl-ds pgm_sts psds;
        thisPgm *proc;
       End-ds;

       Dcl-Proc Proc_Name;

       Dcl-Pi Proc_Name;

       End-Pi;

       Dcl-s abnormal_End Ind Inz(*Off);

       Return;

       On-Exit abnormal_End;

        If abnormal_End;
          DSPLY (thisPgm + 'Errored out.');
        Endif;

       End-Proc Proc_Name;

**Reply by worksofliam** [reply]
This is also the expected functionality.

**Reply by Gregory1701** [reply]
Expected functionality?  So it's not going to support non-**FREE RPG, but still free format RPG?  If this is the direction, you just alienated 90+% of RPG shops.

**Reply by worksofliam** [reply]
Currently, for the time being, references will only support total free format. Trust me, I'm aware of the 'alienation'. 

Sadly, there is only so much time to work on this and it isn't my full time job 😅

**Reply by Gregory1701** [reply]
Okay, well thank you for the explanation.
